serverless/internal/storage/fs/fs_test.go

```diff
@@ -116,3 +117,44 @@ func TestUpdateState(t *testing.T) {
 		t.Errorf("Updated state had diff %s", diff)
 	}
 }
+
+func TestSequence(t *testing.T) {
+	empty := []byte("empty")
+
+	d := filepath.Join(t.TempDir(), "storage")
+	s, err := Create(d, empty)
+	if err != nil {
+		t.Fatalf("Create = %v", err)
+	}
+
+	for _, test := range []struct {
+		desc    string
+		leaves  [][]byte
+		wantErr bool
+	}{
+		{
+			desc:    "sequences ok",
+			leaves:  [][]byte{{0x00}, {0x01}, {0x02}},
+			wantErr: false,
+		}, {
+			desc:    "dupe denied",
+			leaves:  [][]byte{{0x10}, {0x10}},
+			wantErr: true,
```

Comment:
Possibly worth changing this or adding another bool to control checking it's an `os.IsExist` as that's part of the contract calling code relies on.

Reply:
Yeah, done - I've also made an explicit error for `Dupe Leaf` to avoid any ambiguity.